crates/ursa-index-provider/src/provider.rs

```diff
@@ -190,16 +198,25 @@ where
                 current_head.map(|h| forest_ipld::Ipld::Link(convert_cid(h.to_bytes())));
             let sig = ad.sign(&keypair)?;
             ad.Signature = Ipld::Bytes(sig.into_protobuf_encoding());
-            let ipld_ad = forest_ipld::to_ipld(ad)?;
+            let ipld_ad = forest_ipld::to_ipld(&ad)?;
             let cid = bs.put_obj(&ipld_ad, Code::Blake2b256)?;
             *head = Some(cid);
             return Ok(());
         }
         return Err(anyhow!("ad not found"));
     }
 
-    async fn announce_msg(&self, peer_id: PeerId) -> Result<Vec<u8>> {
-        let msg_addrs = [self.config.domain.clone().parse().unwrap()].to_vec();
+    async fn create_announce_msg(&self, peer_id: PeerId) -> Result<Vec<u8>> {
+        let domain = match Multiaddr::from_str(&self.config.domain) {
+            Ok(m) => m,
+            Err(e) => {
+                warn!("The data will not be indexed by indexer.
+                Domain for the index provider is not provided in the config or the provided domain is not a multiaddr.");
+                return Err(anyhow!(e));
+            }
+        };
```

Comment:
```suggestion
        let multi_address = match Multiaddr::from_str(&self.config.domain);
```
just suggestion on the name, as we are returning a multiaddress not a domain